src/test/java/org/assertj/core/internal/classes/Classes_assertHasMethods_Test.java

```diff
@@ -46,20 +49,25 @@ public void should_pass_if_actual_has_expected_accessible_public_methods() {
 
   @Test
   public void should_fail_if_no_methods_are_expected_and_methods_are_available() {
+    TreeSet<String> expectedMethods = newTreeSet("publicMethod",
+                                                 "protectedMethod",
+                                                 "privateMethod",
+                                                 "finalize",
+                                                 "wait",
+                                                 "equals",
+                                                 "toString",
+                                                 "hashCode",
+                                                 "getClass",
+                                                 "clone",
+                                                 "notify",
+                                                 "notifyAll");
+    BigDecimal javaVersion = new BigDecimal(System.getProperty("java.specification.version"));
+    if (javaVersion.compareTo(new BigDecimal("14")) < 0) {
+      expectedMethods.add("registerNatives");
+    }
     assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> classes.assertHasMethods(someInfo(), actual))
```

Comment:
would be nice to use either `AssertionUtil.expectAssertionError` or `AssertionUtil.assertThatAssertionErrorIsThrownBy`